proposals/CSP.md

```diff
@@ -132,65 +132,26 @@ We recommend that a new CSP policy directive `wasm-unsafe-eval` be created. If s
-WebAssembly.validate(bytes)                      | allow | allow | allow | allow 
-new WebAssembly.Instance(module)                 | allow | allow | allow | allow 
-new WebAssembly.CompileError                     | allow | allow | allow | allow 
-new WebAssembly.LinkError                        | allow | allow | allow | allow 
-new WebAssembly.Table                            | allow | allow | allow | allow 
-new WebAssembly.Memory                           | allow | allow | allow | allow 
+* If the `wasm-unsafe-eval` source keyword is used, and the `unsafe-eval` keyword is not present, then WebAssembly modules may be compiled and instantiated.
 
-Where script-src means rejecting operations that are not allowed by the CSP
-policy's directives for the source of scripts, e.g. script-src restricting origins.
-Note that `unsafe-eval` effectively *implies* `wasm-unsafe-eval`; this is, in part, to ensure continued behavior from the current status where `unsafe-eval` is used in order to ensure execution of WebAssembly code. Similarly, applying regular `script-src` policy to WebAssembly is an extension of the existing semantics for `script-src` to a new domain (WebAssembly in addition to JavaScript execution).
+  One advantage of this is that a website can permit WebAssembly modules to be used without also enabling JavaScript's `eval` keyword.
 
-Note that some of these operations (`WebAssembly.instantiate(Module)` and `new WebAssembly.Instance(module)`) are effectively gated by the requirement to construct modules, i.e., these operations are dependent on whether `WebAssembly.compile(bytes)` or `WebAssembly.compileStreaming(Response)` are allowed.
+On the event of failure, then a `CompileError` should be thrown by the `WebAssembly.compile` and related API calls.
 
-On the event of failure, then a `CompileError` should be thrown by the appropriate JavaScript operation.
-
-#### Examples
-
-```
-Content-Security-Policy: script-src 'self';
+### Using existing CSP script-src policies
 
-WebAssembly.compileStreaming(fetch('/foo.wasm'));  // BAD: wasm-unsafe-eval compile error
-WebAssembly.instantiateStreaming(fetch('/foo.wasm')); // BAD: wasm-unsafe-eval compile error
-WebAssembly.compileStreaming(fetch('/foo.js'));  // BAD: mime type
-WebAssembly.instantiateStreaming(fetch('/foo.js')); // BAD: mime type
-WebAssembly.compileStreaming(fetch('http://yo.com/foo.wasm'));  // BAD: cross origin
-WebAssembly.instantiateStreaming(fetch('http://yo.com/foo.wasm')); // BAD: cross origin
-```
+Currently, WebAssembly modules are anomolous within the Web platform because there is no specific HTML element tag that references WebAssembly modules. It is reasonable to suggest that the existing `script` tag may one day be extended to also include WebAssembly modules. In that event, it is also reasonable to consider the use of the CSP `script-src` source keyword to include WebAssembly as well as JavaScript.
```

Comment:
```suggestion
Currently, WebAssembly modules are anomolous within the Web platform because there is no specific HTML element that references WebAssembly modules. It is reasonable to suggest that the existing `script` element may one day be extended to also include WebAssembly modules. In that event, it is also reasonable to consider the use of the CSP `script-src` source keyword to include WebAssembly as well as JavaScript.
```